apps/openstmap/app.js

```diff
@@ -67,16 +69,35 @@ function drawPOI() {
 
 // Draw the marker for where we are
 function drawMarker() {
-  if (!fix.fix) return;
+  if (!fix.fix || !settings.drawMarker) return;
   var p = m.latLonToXY(fix.lat, fix.lon);
   g.setColor(1,0,0);
   g.fillRect(p.x-2, p.y-2, p.x+2, p.y+2);
 }
 
+// Draw current location with LCD Overlay (Bangle.js 2 only)
+function drawLocation() {
+  if (!Bangle.setLCDOverlay) {
+    return; // Overlay not supported
+  }
+
+  if (!fix.fix || !mapVisible) {
+    Bangle.setLCDOverlay(); // clear if map is not visible or no fix
```

Comment:
This seems undocumented but works on my
machine..

Reply:
Yes, that's fine but I think it may force a full redraw of the screen each time it's called, so ideally do some checking first to avoid calling it repeatedly if you know you had't added an overlay?

Reply:
done